internal/util/kubernetes/unstructured_util.go

```diff
@@ -339,3 +294,64 @@ func (obj *GenericObject) DeepCopy() *GenericObject {
 	result.Status = *obj.Status.DeepCopy()
 	return result
 }
+
+func CreateResource(ctx context.Context, c client.Client, obj *GenericObject) error {
+	unstructuredObj, err := ObjectToUnstructured(obj)
+	if err != nil {
+		return err
+	}
+
+	return c.Create(ctx, unstructuredObj)
+}
+
+// GetResource retrieves the resource from the informer cache, if it's not found then it fetches from the API server.
+func GetResource(ctx context.Context, c client.Client, obj *GenericObject) (*GenericObject, error) {
+	unstructuredObj := &unstructured.Unstructured{}
+	unstructuredObj.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+
+	if err := c.Get(ctx, k8stypes.NamespacedName{Name: obj.Name, Namespace: obj.Namespace}, unstructuredObj); err != nil {
+		return nil, err
+	}
+
+	return UnstructuredToObject(unstructuredObj)
+}
+
+func UpdateResource(ctx context.Context, c client.Client, obj *GenericObject) error {
+	unstructuredObj, err := ObjectToUnstructured(obj)
+	if err != nil {
+		return err
+	}
+
+	if err = c.Update(ctx, unstructuredObj); err != nil {
+		return err
+	} else {
+		result, err := UnstructuredToObject(unstructuredObj)
+		if err != nil {
+			return err
+		}
+		*obj = *result
+	}
+
+	return nil
+}
+
+// ListResources retrieves the list of resources from the informer cache, if it's not found then it fetches from the API server.
+func ListResources(ctx context.Context, c client.Client, obj *GenericObject, opts ...client.ListOption) ([]*GenericObject, error) {
```

Comment:
I'm not sure if we did this somewhere else in the code, but I'd prefer to not pass in `obj *GenericObject` - I'd like that kind of parameter to truly represent an object as opposed to just being somewhere to hold a `GroupVersionKind`

Reply:
the `GetResource()` and `GetCR()` functions are kind of like that too - the `GenericObject` is where the name and namespace are coming from, but theoretically we're not actually using the object itself - could consider changing that too, although I know it's been like that for the history of this codebase

Reply:
How about having it as `gvk schema.GroupVersionKind` instead of `obj *GenericObject` ? like below
```
ListResources(ctx context.Context, c client.Client, gvk schema.GroupVersionKind, opts ...client.ListOption)
```

Reply:
perfect!